tests/linux/test_local_network_access_manager.cpp

```diff
@@ -244,6 +287,77 @@ TEST_F(LocalNetworkAccessManager, query_in_url_is_preserved)
     handle_request(base_url, "GET");
 }
 
+TEST_F(LocalNetworkAccessManager, sending_chunked_data_receives_expected_data)
+{
+    QByteArray random_data = generate_random_data(65536);
+    QByteArray http_response{"HTTP/1.1 200 OK\r\n\r\n"};
+
+    auto server_response = [&http_response, &random_data](auto data) {
+        QByteArray first_part = random_data.mid(0, 32768);
+        QByteArray second_part = random_data.mid(32768);
+
+        // Implies the data was split, ie, chunked
+        EXPECT_FALSE(data.contains(random_data));
+
+        // Implies that the correct data was still received
+        EXPECT_TRUE(data.contains(first_part));
+        EXPECT_TRUE(data.contains(second_part));
```

Comment:
`.contains` feels like a weak test, taking into account the `\r\n` doubt I have above?

Reply:
I kind of thought this test is not the best, but the alternative is having to implement a "server" that can parse different lengths of data transfers, etc. and I feel that is starting to take away from the actual implementation of what we are trying to accomplish.  That said, I'll look more into this and see if there is a better way.

Reply:
As mentioned below, I really don't think I should be the one validating the very HTTP that I'm creating in `LocalSocketReply::send_request()` as any misunderstandings of the spec will likely transfer over into the unit test validation.

For now, I'm going to add a TODO to figure out a better way to validate HTTP in the long term and in the short term just make it so coverage is satisfied.  WDYT?